caddyhttp/log/log.go

```diff
@@ -66,6 +67,14 @@ func (l Logger) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 
 			// Write log entries
 			for _, e := range rule.Entries {
+				if e.Log.IPMaskExists {
+					// Mask IP Address, will not modify if default
+					hostip, hostport, err := net.SplitHostPort(r.RemoteAddr)
+					if err == nil {
+						r.RemoteAddr = net.JoinHostPort(e.Log.MaskIP(hostip), hostport)
```

Comment:
Just a concern how to get the real `RemoteAddr` later if we replace it here.

Reply:
I made an assumption, very possibly wrong that Log is the last thing to be called.  If not I can look at a way around this.

Reply:
I think it's unlikely that log is the last middleware in the chain.